air/src/execution_step/instructions/fold_stream_map.rs

```diff
@@ -0,0 +1,147 @@
+/*
```

Comment:
Can we avoid duplicating code here and use an underlying stream instead?

Reply:
Any suggestions how to proceed from here?